types/logger.d.ts

```diff
@@ -1,6 +1,7 @@
 import { FastifyError } from 'fastify-error'
 import { RawServerBase, RawServerDefault, RawRequestDefaultExpression, RawReplyDefaultExpression } from './utils'
 import { FastifyRequest, RequestGenericInterface } from './request'
+import { PrettyOptions } from 'pino'
```

Comment:
I'm a bit worried about TypeScript users. Will it work when user does not have `@types/pino` package installed?

Reply:
Indeed!

Reply:
`pino` does not provides own types and they shipped via `@types/pino` package hence If I won't install `@types/pino` TypeScript will consider typings as invalid

Reply:
So would you prefer to just copy the type definition over?  That way would mean it would have to be updated here on changes from `pino` / `@types/pino`.

Alternatively a note could be added to the docs.  In order to use `pino-pretty` with `fastify` you need to install the `pino-pretty` dependency regardless so in the docs a note can be added to also install `@types/pino` if you are using TypeScript.

Reply:
I would include typings in `pino` itself.
BTW moving `@types/pino` to `dependencies` or adding notes to README also acceptable, but as a WebStorm IDE user I would prefer first (moving `@types/pino`) for better smart completion.

Reply:
If team agrees, I can submit PR for type inclusion in pino.

Reply:
I'm +1

Reply:
I don't like including types that are not under our control (from `@types/pino`) to the fastify dependency tree.
@mcollina what do you think? What about a PR to pino itself to add typings? @jsumners any thoughts?

Reply:
I think this has been already discussed in past and @fox1t get the point

Here one quick answer:
https://github.com/fastify/fastify/pull/1532#discussion_r265271715

Reply:
We have decided in the past not to ship the pino types with our module. We'd like it if there were instructions to install pino types if the user plans on using the pino logger. Otherwise, the logger types we provide should be the minimum types necessary for using the logger as is.

Reply:
Ok. So now we have one remaining option: since pino is already a dependency of fastify, if pino had integrated types we could have them out of the box. Is any particular reason why pino relies on `@types/`.

Reply:
Pino has no official types. No one of the maintainers of pino has enough typescript skills to have types. The types on `@types` are unofficial and should not be used.

Reply:
@mcollina I agree with you. Do you want me to contribute to pino types and maintain them in the repo itself? I think that there are other folks here that can help to maintain them since pino is THE logger of Fastify.

Reply:
@fox1t Would be happy to volunteer for helping with official pino types.

Reply:
@fox1t i can help too

Reply:
I don't like including types that are not under our control (from `@types/pino`) to the fastify dependency tree.
@mcollina what do you think? What about a PR to pino itself to add typings? @jsumners any thoughts?

Reply:
Any thoughts?

Reply:
I would open an issue on pino, offering to write and maintain them in the long run and see it from there.

Reply:
Done: https://github.com/pinojs/pino/issues/910

Reply:
> I don't like including types that are not under our control (from `@types/pino`) to the fastify dependency tree.
> @mcollina what do you think? What about a PR to pino itself to add typings? @jsumners any thoughts?

As you've learned, I 1. have these TS issues muted and 2. I have opinions. Opinions you will not appreciate.

Reply:
@jsumners I appreciate very much your opinions actually. Even more, if your opinions are different from mine. That's why I tagged you here and why I ask your point of view whenever I can. 

However, if you don't want to be tagged on TS issues, I am not going to tag you in the future.

Reply:
That would be fine. I cannot provide any useful input for TS related things. If you want the full story, read through https://github.com/fastify/fastify/issues/649

Reply:
I've just finished reading it. That issue was a really long ride.

Reply:
Indeed.